### ISSUE TITLE
Zig doesn't make PDB debug files correctly on Windows?

### ISSUE THREAD

**Issue by heavymetalmixer** [ziglang/zig#22039]
### Zig Version

0.13.0

### The following are steps to reproduce the issue, which is not being able to debug Zig binaries on Windows with PDB files, and workarounds to make the debugging possible. I tested them on Windows 10 Pro x64 in VS Code, but the first link comes from someone who tried them on NeoVim.

**Step 1:** Install CodeLLDB extension in VS Code, it integrates the LLDB debugger in VS Code to debug C, C++ and Zig files (maybe more from LLVM).

**Step 2:** Create the .vscode folder inside the project's folder and create a "launch.json file" inside that folder.

**Step 3:** Paste the following inside the launch.json file and save it:

```
{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "CodeLLDB Debug",
            "type": "lldb",
            "request": "launch",
            "program": "${workspaceFolder}/zig-out/bin/hello-world.exe",
            "args": [],
            "cwd": "${workspaceFolder}",
            "stopOnEntry": false,
            "terminal": "integrated",
        }
    ]
}
```

Step 4: Try to run the debugger and see that it ignores all breakpoints and only runs the binary.

If you set "stopOnEntry" to true, then VS Code will show you "Exception 0x80000003" between assembly code and the debugger will crash after a few debugging steps.

Note: The same issue happens on Zig 0.12.0 and 0.12.1. I didn't test it with older versions than those.

![z1](https://github.com/user-attachments/assets/cced7abf-1d54-4b03-8237-4be413f5900b)
![z2](https://github.com/user-attachments/assets/4889fe99-b8f6-43bf-a193-feac159db645)

---------------------------------------------------------------------------------------------------------------------------------

### Workarounds

1) Create the environment variable for your system ```LLDB_USE_NATIVE_PDB_READER``` and set it to 1. Restart VSCode, and CodeLLDB should be able to debug Zig binaries with DPB debug files as intented.

### Given that this workaround isn't necessary for debugging C and C++ files it makes me think that Zig creates "broken" PDB files or in a non-standard way.

Both links have a lot of info on how LLDB works on VSCode, I recommend you read them.

Sources: 
https://www.reddit.com/r/neovim/comments/1eeydfp/solving_codelldb_with_dap_not_picking_up_pdb/
https://werat.dev/blog/debugging-wine-with-lldb-and-vscode/

--------------------------------------------------------------------------------------------------------------------------------

2) The other workaround is to use the Microsoft debugger instead by using the following inside the launch.json file:

```
{
      "name": "MSVC Debug",
      "type": "cppvsdbg",
      "request": "launch",
      "program": "${workspaceFolder}/zig-out/bin/<executable name>.exe",
      "args": [],
      "stopAtEntry": false,
      "cwd": "${workspaceFolder}",
      "environment": [],
      "console": "internalConsole",
}
```

### Expected Behavior

LLDB should be able to debug Zig binaries on Windows without issues nor having to use workarounds. On VS Code specifically, when using the CodeLLDB extension or when using LLDB manually from the console.

It would also be nice if Zig had an option to create DWARF files instead of PDB ones to be able to debug the binaries with GDB.

**Comment by heavymetalmixer** [comment]
One more "bug" discovered (I don't if it's really a bug or intented behavior):

**Both LLDB and MSVC Debugger ignore imports.**

**Comment by heavymetalmixer** [comment]
Update on this issue: After updating LLVM from 19.1.0 to 19.1.4 there was a change. 

With the environment variable LLDB_USE_NATIVE_PDB_READER set it to 1 the LLDB behavior changed Zig. Instead of always jumping to the next line of code, it only jumps if the breakpoint is set on top of a function name or a comment.

**Comment by kiffpuppygames** [comment]
For anyone who is looking... If you are trying to debug zig with lldb in VSCodium this will work.

**Comment by heavymetalmixer** [comment]
> For anyone who is looking... If you are trying to debug zig with lldb in VSCodium this will work.

On Windows?

**Comment by kiffpuppygames** [comment]
Yes!

Sent from Proton Mail Android

-------- Original Message --------
On 01/02/2025 02:30, heavymetalmixer wrote:
For anyone who is looking... If you are trying to debug zig with lldb in VSCodium this will work.
On Windows?

**Comment by heavymetalmixer** [comment]
> Yes!

That's interesting. Are the extensions on both editors different?
Btw, does VS Codium have a Zig extension?

**Comment by kiffpuppygames** [comment]
Yup, there is a zig extension and the version on both stores are the same as well as the change log. Same for CodeLLDB. I also had a look at the git repos and in both cases they appear to be deployed from the same repo.
